pdr_backend/pdr_dashboard/callbacks/callbacks_common.py

```diff
@@ -7,10 +7,26 @@
 from pdr_backend.pdr_dashboard.pages.feeds import FeedsPage
 from pdr_backend.pdr_dashboard.pages.home import HomePage
 from pdr_backend.pdr_dashboard.pages.predictoors import PredictoorsPage
-from pdr_backend.pdr_dashboard.util.helpers import get_date_period_text_header
+from pdr_backend.pdr_dashboard.util.helpers import (
+    get_date_period_text_header,
+    with_loading,
+    check_data_loaded,
+)
 
 
 def get_callbacks_common(app):
+    @app.callback(
+        Output("is-initial-data-loaded", "data"),
+        Input("is-initial-data-loaded", "data"),
+    )
+    def initial_data_load(is_data_loaded):
```

Comment:
I think this thread has some pertinent options: https://community.plotly.com/t/is-there-a-way-to-trigger-load-on-initial-page-load-only-and-not-every-time-a-change-is-made-to-the-page/57504